caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -728,18 +785,19 @@ def _truncate_input_tokens(
                         f"maximum sequence length for this model ({tokens} > {max_tokens})."
                     ),
                 )
+        input_token_count = self._sum_token_count(tokenized)
 
-        return tokenized
+        return tokenized, input_token_count
 
     def encode(
         self,
-        sentences: Union[str, List[str]],
+        sentences: Union[str, Collection[str]],
```

Comment:
We already have a signature does not match the base class situation (added an arg we need and chose to leave some out that we ignore), but I don't see much value in these new tweaks.  Were we already abusing the hints on those?

Reply:
It was an attempt to follow this [guidance](https://typing.readthedocs.io/en/latest/source/best_practices.html#arguments-and-return-types) which states to use abstract base classes for arguments and concrete types for return types.

I'm reverting this to `List[str]` since I also personally find it easier to read (the specifics of ABC's are not widely known).